src/android/MainActivity.java

```diff
@@ -0,0 +1,652 @@
+            recorder.setVideoEncoder(MediaRecorder.VideoEncoder.H264);
+            recorder.setAudioEncoder(MediaRecorder.AudioEncoder.AAC);
+            recorder.setAudioEncodingBitRate(profile.audioBitRate);
+            recorder.setAudioSamplingRate(profile.audioSampleRate);
+            int rotation = this.getWindowManager().getDefaultDisplay().getRotation();
+            switch (sensorOrientation) {
+                case SENSOR_ORIENTATION_DEFAULT_DEGREES:
+                    recorder.setOrientationHint(DEFAULT_ORIENTATIONS.get(rotation));
+                    break;
+                case SENSOR_ORIENTATION_INVERSE_DEGREES:
+                    recorder.setOrientationHint(INVERSE_ORIENTATIONS.get(rotation));
+                    break;
+            }
+
+            recorder.prepare();
+
+        } catch (IOException e) {
+            e.printStackTrace();
+
+        }
+    }
+
+
+    private void startBackgroundThread() {
+        backgroundThread = new HandlerThread("CameraBackground");
+        backgroundThread.start();
+        backgroundHandler = new Handler(backgroundThread.getLooper());
+    }
+
+    public void cancelActivity() {
+
+        if (isRecordingVideo) {
+            File file = new File(lastRecordedFileUrl);
+            file.delete();
+        }
+        finish();
+    }
+    public void startRecordingVideo() {
+        if (isRecordingVideo) {
+            return;
+        }
+        if (null == cameraDevice || !textureView.isAvailable() || null == previewsize) {
+            return;
+        }
+        try {
+            closePreviewSession();
+            prepareMediaRecorder();
+            SurfaceTexture texture = textureView.getSurfaceTexture();
+            assert texture != null;
+            texture.setDefaultBufferSize(previewsize.getWidth(), previewsize.getHeight());
+            previewBuilder = cameraDevice.createCaptureRequest(CameraDevice.TEMPLATE_RECORD);
+            List<Surface> surfaces = new ArrayList<>();
+
+            Surface previewSurface = new Surface(texture);
+            surfaces.add(previewSurface);
+            previewBuilder.addTarget(previewSurface);
+            //MediaRecorder setup for surface
+            Surface recorderSurface = recorder.getSurface();
+            surfaces.add(recorderSurface);
+            previewBuilder.addTarget(recorderSurface);
+            // Start a capture session
+            cameraDevice.createCaptureSession(surfaces, new CameraCaptureSession.StateCallback() {
+                @Override
+                public void onConfigured( CameraCaptureSession cameraCaptureSession) {
+                    previewSession = cameraCaptureSession;
+                    updatePreview();
+
+                    MainActivity.this.runOnUiThread(() -> {
+                        isRecordingVideo = true;
+                        // Start recording
+                        recorder.start();
+                        lockDeviceRotation(true);
```

Comment:
idk if that matter but should we first lock rotation and then start recording. meaning moving this line one up

Reply:
fixed